### ISSUE TITLE
Question about defining _get_observations() in DirectRLEnvCfg

### ISSUE THREAD

**Issue by Jushan-Randy-Chen** [isaac-sim/IsaacLab#2714]
Hi,

I was going through the "walkthrough" section on the Isaac lab documentation page. In the "Exploring the RL Problem" section they define a `get_observations()` function for the cart-pole environment:

```
def _get_observations(self) -> dict:
    self.velocity = self.robot.data.root_com_vel_w
    self.forwards = math_utils.quat_apply(self.robot.data.root_link_quat_w, self.robot.data.FORWARD_VEC_B)

    dot = torch.sum(self.forwards * self.commands, dim=-1, keepdim=True)
    cross = torch.cross(self.forwards, self.commands, dim=-1)[:,-1].reshape(-1,1)
    forward_speed = self.robot.data.root_com_lin_vel_b[:,0].reshape(-1,1)
    obs = torch.hstack((dot, cross, forward_speed))

    observations = {"policy": obs}
    return observations
```

However, I see that this method `_get_observations()` is not called when computing the rewards under `_get_rewards(self) `. My question is, why do we even define this `_get_observations()` function in the first place? If we really have a partially observable environment, we should only define the reward based on the parameters returned by "_get_observations"

Thanks

**Comment by HritikRC** [comment]
Hi @Jushan-Randy-Chen,

I hope I can answer this, though I am not an Isaac Lab contributor/developer. 

Essentially, the observations returned by the `_get_observations` method can be used as inputs to the policy/value network which gets trained (check the YAML files of whatever RL library the project is created with). The output of this network are of course the actions. So, if you don't define the `_get_observations` method, there are no observations being created, and so, the agent will not receive any observations as inputs to its network.

The rewards are computed separately. The `_get_observations` and `_get_rewards` methods are "built-in" methods from the DirectRLEnv class which are called automatically with Isaac Lab, so don't worry about calling these. 

Hope this helps!
